ui-react/src/components/topBar/TopBar.tsx

```diff
@@ -1,16 +1,54 @@
-import React from 'react';
-import NavBar from 'react-bootstrap/Navbar';
-import { Container } from 'react-bootstrap';
-import logo from '../../assets/logoFishOnly.svg';
+import React, { useState } from "react";
+import NavBar from "react-bootstrap/Navbar";
+import {
+  Container,
+  Stack,
+} from "react-bootstrap";
+import logo from "../../assets/logoFishOnly.svg";
+import { openExternal } from "../../eventsBridge";
+import { repository } from "../../../package.json";
+import { discordchannel } from "../../../package.json";
+import NavButton from "./NavButton";
+import styles from "./TopBar.module.css";
+import ReportProblemModal from "./ReportProblemModal";
+import Image from 'react-bootstrap/Image'
 
 function TopBar() {
+  const [showReportModal, setShowReportModal] = useState(false);
+
+  const handleCloseModal = () => setShowReportModal(false);
+
+  const handleOpenGithub = () => {
+    openExternal(repository);
+  };
+
+  const handleDiscord = () => {
+    openExternal(discordchannel);
+  };
+
+  const handleReportProblem = () => {
+    setShowReportModal(true);
+  };
```

Comment:
If it is just a function calling to another function, I think you can write the real function just inside the JSX and save us from reading another step just to understand what is append:

`<NavButton onClick={() => setShowReportModal(true)} text="דיווח על בעיה" />`